frontend/src/redux/configureStore.js

```diff
@@ -1,15 +1,15 @@
 // @flow
 import { createStore, applyMiddleware, compose } from 'redux'
-import { routerMiddleware } from 'react-router-redux'
-import createHistory from 'history/createBrowserHistory'
+import { routerMiddleware } from 'connected-react-router'
+import { createBrowserHistory } from 'history'
 import thunk from 'redux-thunk'
-import reducer from './modules/reducer'
+import createRootReducer from './modules/reducer'
 
-export const history = createHistory()
+export const history = createBrowserHistory()
 const middleware = [thunk, routerMiddleware(history)]
 const composedMiddleware = compose(applyMiddleware(...middleware))
 const devtools = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
 
-export default function configureStore() {
-  return createStore(reducer, devtools(composedMiddleware))
+export default function configureStore(preloadedState) {

```

Comment:
I'd remove `preloadedState` from here. Since we call `configureStore` from our outermost React file, that file has no need to retrieve the initial Redux state. Usually when we do have an initial state, it would be from Local Storage, or some other Browser API, so we wouldn't pass it in through here.